### PR TITLE
fix(common): add support for Gateway API v1

### PR DESCRIPTION
**Pull request by larivierec**
### Description of the change

Simply adds gateway apis v1 support.
Some of the resources GA'ed

#### Added

Allows us to be able to create v1 resources.
- HTTPRoute route is able to be created using the natural v1
- other's will follow as they are GA'ed (but currently this is the only one)

Gateways and GatewayClasses, which are not supported via the helm chart are the other two.

### Benefits

- Support of wider range of types, v1apha2 (which will eventually be removed)
- Support of v1beta
- Support of v1

### Possible drawbacks

None

### Applicable issues

None

## Checklist

- [X] Title of the PR conforms to the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
- [X] Scope of the of the PR title contains the chart name.
- [X] Chart version in `Chart.yaml` has been bumped according to [Semantic Versioning](https://semver.org/).
- [X] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [-] Variables have been documented in the `values.yaml` file.

### CODE REVIEW

**Comment by larivierec** [comment]
hi @bjw-s I know this isn't the biggest of PR's but could you take a look at this PR?
I'm not sure how to request a review and considering you own this repo I thought you'd be the right person.

I think currently that v1beta1 is upwards converting for `Gateway`, `GatewayClass` and `HTTPRoute` but it would be good to have this in.

**Comment by Goorzhel** [comment]
> Title of the PR conforms to the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
> Scope of the PR title contains the chart name.

You're gonna wanna retitle the commit and PR to something like

> feat(common): add support for Gateway API v1

File: charts/library/common/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: v2
 name: common
 description: Function library for Helm charts
 type: library
-version: 2.0.3
+version: 2.0.4
```
Comment (bjw-s) [inline charts/library/common/Chart.yaml:6]:
```suggestion
version: 2.1.0
```

File: charts/library/common/Chart.yaml [hunk 2]
```diff
@@ -16,4 +16,4 @@ annotations:
   artifacthub.io/changes: |-
     - kind: fixed
```
Comment (bjw-s) [inline charts/library/common/Chart.yaml:17]:
```suggestion
    - kind: added
```

**Comment by larivierec** [comment]
Thanks both, I'll know for next time!

**Pull request merged**